src/service/action-impl.js

```diff
@@ -412,85 +445,97 @@ export class ActionService {
   action_(source, actionEventType, event, trust) {
     const action = this.findAction_(source, actionEventType);
     if (!action) {
-      // TODO(dvoytenko): implement default (catch-all) actions.
       return;
     }
-
     // Invoke actions serially, where each action waits for its predecessor
     // to complete. `currentPromise` is the i'th promise in the chain.
     let currentPromise = null;
-
     action.actionInfos.forEach((actionInfo, i) => {
+      const targetType = actionInfo.target;
       // Replace any variables in args with data in `event`.
       const args = dereferenceExprsInArgs(actionInfo.args, event);
-
-      const invoke = () => {
-        // Global target, e.g. `AMP`.
-        const globalTarget = this.globalTargets_[actionInfo.target];
-        if (globalTarget) {
-          const invocation = new ActionInvocation(this.root_, actionInfo.method,
-              args, source, action.node, event, trust);
-          return globalTarget(invocation, i, action.actionInfos);
```

Comment:
I moved this to `invoke_` so now all actions are invoked through that function. This is more readable and means I can check the whitelist in single place.